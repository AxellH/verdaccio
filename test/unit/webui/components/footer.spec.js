
import React from 'react';
import { mount } from 'enzyme';
import { IntlProvider } from 'react-intl';

import Footer from '../../../../src/webui/components/Footer/index';

describe('<Footer /> component', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = mount(
            <IntlProvider locale='en'>
                <Footer />
            </IntlProvider>,
        );
    });

    it('should set html from props poweredBy', () => {
        expect(
          wrapper
            .find('#poweredBy')
            .text()
        ).toEqual('Powered by');
        expect(wrapper.html()).toMatchSnapshot();
      });

    it('should set html from props withLove', () => {
        expect(
          wrapper
            .find('#withLove')
            .text()
        ).toEqual('Made with ❤ on');
        expect(wrapper.html()).toMatchSnapshot();
      });

    it('should load the inital state of Footer component', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('should hover the earth in footer', () => {
        const earth = wrapper.find('img[alt="Earth"]');
        earth.simulate('click');
        setTimeout(function() {
            expect(wrapper.state('showAuthorsGeographic')).toEqual(true);
        }, 100);
    });
});
