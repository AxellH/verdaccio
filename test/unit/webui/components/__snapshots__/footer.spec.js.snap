// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> component should load the inital state of Footer component 1`] = `"<div class=\\"wrap\\"><footer class=\\"container footer false\\"><span id=\\"withLove\\">Made with ❤ on</span><img class=\\"earth emoji\\" src=\\"[object Object]\\" alt=\\"Earth\\"><div class=\\"tooltip\\"><img src=\\"[object Object]\\" alt=\\"Brazil\\" title=\\"Brazil\\" class=\\"emoji\\"><img src=\\"[object Object]\\" alt=\\"China\\" title=\\"China\\" class=\\"emoji\\"><img src=\\"[object Object]\\" alt=\\"India\\" title=\\"India\\" class=\\"emoji\\"><img src=\\"[object Object]\\" alt=\\"Nicaragua\\" title=\\"Nicaragua\\" class=\\"emoji\\"><img src=\\"[object Object]\\" alt=\\"Pakistan\\" title=\\"Pakistan\\" class=\\"emoji\\"><img src=\\"[object Object]\\" alt=\\"Spain\\" title=\\"Spain\\" class=\\"emoji\\"></div><div class=\\"right\\"><span id=\\"poweredBy\\">Powered by</span><a href=\\"http://www.verdaccio.org/\\"><img class=\\"logo\\" src=\\"[object Object]\\" alt=\\"Verdaccio\\" title=\\"Verdaccio\\"></a>&nbsp;/&nbsp;1.0.0</div></footer></div>"`;

exports[`<Footer /> component should set html from props poweredBy 1`] = `"<div class=\\"wrap\\"><footer class=\\"container footer false\\"><span id=\\"withLove\\">Made with ❤ on</span><img class=\\"earth emoji\\" src=\\"[object Object]\\" alt=\\"Earth\\"><div class=\\"tooltip\\"><img src=\\"[object Object]\\" alt=\\"Brazil\\" title=\\"Brazil\\" class=\\"emoji\\"><img src=\\"[object Object]\\" alt=\\"China\\" title=\\"China\\" class=\\"emoji\\"><img src=\\"[object Object]\\" alt=\\"India\\" title=\\"India\\" class=\\"emoji\\"><img src=\\"[object Object]\\" alt=\\"Nicaragua\\" title=\\"Nicaragua\\" class=\\"emoji\\"><img src=\\"[object Object]\\" alt=\\"Pakistan\\" title=\\"Pakistan\\" class=\\"emoji\\"><img src=\\"[object Object]\\" alt=\\"Spain\\" title=\\"Spain\\" class=\\"emoji\\"></div><div class=\\"right\\"><span id=\\"poweredBy\\">Powered by</span><a href=\\"http://www.verdaccio.org/\\"><img class=\\"logo\\" src=\\"[object Object]\\" alt=\\"Verdaccio\\" title=\\"Verdaccio\\"></a>&nbsp;/&nbsp;1.0.0</div></footer></div>"`;

exports[`<Footer /> component should set html from props withLove 1`] = `"<div class=\\"wrap\\"><footer class=\\"container footer false\\"><span id=\\"withLove\\">Made with ❤ on</span><img class=\\"earth emoji\\" src=\\"[object Object]\\" alt=\\"Earth\\"><div class=\\"tooltip\\"><img src=\\"[object Object]\\" alt=\\"Brazil\\" title=\\"Brazil\\" class=\\"emoji\\"><img src=\\"[object Object]\\" alt=\\"China\\" title=\\"China\\" class=\\"emoji\\"><img src=\\"[object Object]\\" alt=\\"India\\" title=\\"India\\" class=\\"emoji\\"><img src=\\"[object Object]\\" alt=\\"Nicaragua\\" title=\\"Nicaragua\\" class=\\"emoji\\"><img src=\\"[object Object]\\" alt=\\"Pakistan\\" title=\\"Pakistan\\" class=\\"emoji\\"><img src=\\"[object Object]\\" alt=\\"Spain\\" title=\\"Spain\\" class=\\"emoji\\"></div><div class=\\"right\\"><span id=\\"poweredBy\\">Powered by</span><a href=\\"http://www.verdaccio.org/\\"><img class=\\"logo\\" src=\\"[object Object]\\" alt=\\"Verdaccio\\" title=\\"Verdaccio\\"></a>&nbsp;/&nbsp;1.0.0</div></footer></div>"`;
